infra/src/main/kotlin/fi/fta/geoviite/infra/publication/PublicationValidationService.kt

```diff
@@ -329,12 +319,30 @@ constructor(
     private fun validateTrackNumber(
         id: IntId<LayoutTrackNumber>,
         validationContext: ValidationContext,
-    ): List<LayoutValidationIssue>? =
-        validationContext.getTrackNumber(id)?.let { trackNumber ->
-            val kmPosts = validationContext.getKmPostsByTrackNumber(id)
-            val referenceLine = validationContext.getReferenceLineByTrackNumber(id)
-            val locationTracks = validationContext.getLocationTracksByTrackNumber(id)
-            val referenceIssues = validateTrackNumberReferences(trackNumber, referenceLine, kmPosts, locationTracks)
+    ): List<LayoutValidationIssue>? {
+        val trackNumber = validationContext.getTrackNumber(id)
+        val kmPosts = validationContext.getKmPostsByTrackNumber(id)
+        val referenceLine = validationContext.getReferenceLineByTrackNumber(id)
+        val locationTracks = validationContext.getLocationTracksByTrackNumber(id)
+        val trackNumberIsCancelled = validationContext.trackNumberIsCancelled(id)
+
+        if (trackNumber == null) {
```

Comment:
Tämä ei mielestäni tarvitsisi tämän tason if-else-rakennetta, vaan `trackNumberExists`:in voisi passata myös tyylillä `trackNumber?.exists ?: false`. Tosin aiemman kommenttini mukaisesti koko `trackNumber`:in voisi vaan passata tuolle validaatiofunkkarille ja tehdä tämän laskennan siellä